soroban-sdk-macros/src/lib.rs

```diff
@@ -244,15 +244,13 @@ pub fn contractimpl(metadata: TokenStream, input: TokenStream) -> TokenStream {
                 #imp
                 #derived_ok
             };
-            if cfg!(feature = "testutils") {
-                let cfs = derive_contract_function_registration_ctor(
-                    crate_path,
-                    ty,
-                    trait_ident,
-                    pub_methods.into_iter(),
-                );
-                output.extend(quote! { #cfs });
-            }
+            let cfs = derive_contract_function_registration_ctor(
```

Comment:
I'm not sure I understand why did we remove the feature gate here.

Reply:
Instead of feature getting the generation off the code, I've changed it to always generate and I've added a cfg feature gate as an attribute on the code. 

I don't know why, I'd guess a rust analyser bug, when we do it the original way it gets out of sync with the function it calls which is always generate and has a cfg feature gate via an attribute.